source/globalCommands.py

```diff
@@ -2279,62 +2283,110 @@ def script_reportOrShowFormattingAtCaret(self, gesture):
-		system focus.
+	def _getNvdaObjWithAnnotationUnderCaret(self) -> Optional[NVDAObject]:
+		"""If it has an annotation, get the NVDA object for the single character under the caret or the object
+		with system focus.
 		@note: It is tempting to try to report any annotation details that exists in the range formed by prior
 			and current location. This would be a new paradigm in NVDA, and may feel natural when moving by line
 			to be able to more quickly have the 'details' reported. However, there may be more than one 'details
 			relation' in that range, and we don't yet have a way for the user to select which one to report.
 			For now, we minimise this risk by only reporting details at the current location.
 		"""
-		log.debug("Report annotation details summary at current location.")
 		try:
 			# Common cases use Caret Position: vbuf available or object supports text range
 			# Eg editable text, or regular web content
 			# Firefox and Chromium support this even in a button within a role=application.
 			caret: textInfos.TextInfo = api.getCaretPosition()
 		except RuntimeError:
 			log.debugWarning("Unable to get the caret position.", exc_info=True)
-			return
+			return None
 		caret.expand(textInfos.UNIT_CHARACTER)
-		nvdaObject: NVDAObject = caret.NVDAObjectAtStart
-		if config.conf["debugLog"]["annotations"]:
-			log.debug(f"Trying with nvdaObject : {nvdaObject}")
-
-		annotation: Optional[str] = nvdaObject.detailsSummary
-		if annotation:
-			log.debug("NVDAObjectAtStart of caret has details.")
+		objAtStart: NVDAObject = caret.NVDAObjectAtStart
+		_isDebugLogCatEnabled = bool(config.conf["debugLog"]["annotations"])
+		if _isDebugLogCatEnabled:
+			log.debug(f"Trying with nvdaObject : {objAtStart}")
+
+		if objAtStart.detailsSummary:
+			log.debug(f"NVDAObjectAtStart of caret has details: {objAtStart.detailsSummary}")
+			return objAtStart
 		elif api.getFocusObject():
 			# If fetching from the caret position fails, try via the focus object
 			# This case is to support where there is no virtual buffer or text interface and a caret position can
 			# not be fetched.
 			# There may still be an object with focus that has details.
 			# There isn't a known test case for this, however there isn't a known downside to attempt this.
 			focus = api.getFocusObject()
-			if config.conf["debugLog"]["annotations"]:
+			if _isDebugLogCatEnabled:
 				log.debug(f"Trying focus object: {focus}")
-			annotation = focus.detailsSummary
-			if annotation:
-				if config.conf["debugLog"]["annotations"]:
+
+			if focus.detailsSummary:
+				if _isDebugLogCatEnabled:
 					log.debug("focus object has details, able to proceed")
+				return focus
 
-		if not annotation:
-			if config.conf["debugLog"]["annotations"]:
-				log.debug("no details annotation found")
+		if _isDebugLogCatEnabled:
+			log.debug("no details annotation found")
+		return None
+
+	_annotationNav = _AnnotationNavigation()

```

Comment:
Don't think this line is needed anymore.

Reply:
It is used in `script_reportDetailsSummary` to manage cycling through annotation targets for a given source